pkg/gardenlet/controller/networkpolicy/controller.go

```diff
@@ -58,8 +59,18 @@ type Controller struct {
 	waitGroup               sync.WaitGroup
 }
 
-// NewController instantiates a new controller.
-func NewController(ctx context.Context, seedClient kubernetes.Interface, seedDefaultNamespaceKubeInformer kubeinformers.SharedInformerFactory, seedLogger *logrus.Entry, recorder record.EventRecorder, seedName string) (*Controller, error) {
+// NewController instantiates a new networkpolicy controller.
+func NewController(ctx context.Context, clientMap clientmap.ClientMap, logger *logrus.Logger, recorder record.EventRecorder, seedName string) (*Controller, error) {
+	seedClient, err := clientMap.GetClient(ctx, keys.ForSeedWithName(seedName))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get garden client: %w", err)
+	}
+
+	// if another controller also needs to work with endpoint resources in the Seed cluster, consider replacing this informer factory with a reusable informer factory.
+	// if this informer factory is not bound to the default namespace, make sure that the endpoint event handlers FilterFunc() also filters for the default namespace.
+	seedDefaultNamespaceKubeInformer := kubeinformers.NewSharedInformerFactoryWithOptions(seedClient.Kubernetes(), 0, kubeinformers.WithNamespace(corev1.NamespaceDefault))
+	endpointsInformer := seedDefaultNamespaceKubeInformer.Core().V1().Endpoints()
```

Comment:
Can we use the informer from the ctrl-runtime here?

Reply:
I adjusted the code to look like the other informers in 3e72e103a. Is that what you meant? :)

Reply:
Yes, thanks! I think there are still some left-overs after this change, see next review! :)